sale_timesheet_task_exclude/tests/test_sale_timesheet_exclude_task.py

```diff
@@ -0,0 +1,129 @@
+        self.SudoSaleOrderLine = self.SaleOrderLine.sudo()
+        self.ProjectCreateSaleOrder = self.env['project.create.sale.order']
+
+    def test_1(self):
+        account = self.SudoAccountAccount.create({
+            'code': 'TEST-1',
+            'name': 'Sales #1',
+            'reconcile': True,
+            'user_type_id': self.user_type_revenue.id,
+        })
+        project = self.SudoProject.create({
+            'name': 'Project #1',
+            'allow_timesheets': True,
+        })
+        product = self.SudoProductProduct.create({
+            'name': 'Service #1',
+            'standard_price': 30,
+            'list_price': 90,
+            'type': 'service',
+            'invoice_policy': 'delivery',
+            'uom_id': self.uom_hour.id,
+            'uom_po_id': self.uom_hour.id,
+            'default_code': 'CODE-1',
+            'service_type': 'timesheet',
+            'service_tracking': 'task_global_project',
+            'project_id': project.id,
+            'taxes_id': False,
+            'property_account_income_id': account.id,
+        })
+        employee = self.SudoEmployee.create({
+            'name': 'Employee #1',
+            'timesheet_cost': 42,
+        })
+        account_payable = self.SudoAccountAccount.create({
+            'code': 'AP1',
+            'name': 'Payable #1',
+            'user_type_id': self.user_type_payable.id,
+            'reconcile': True,
+        })
+        account_receivable = self.SudoAccountAccount.create({
+            'code': 'AR1',
+            'name': 'Receivable #1',
+            'user_type_id': self.user_type_receivable.id,
+            'reconcile': True
+        })
+        partner = self.SudoPartner.create({
+            'name': 'Partner #1',
+            'email': 'partner1@localhost',
+            'customer': True,
+            'property_account_payable_id': account_payable.id,
+            'property_account_receivable_id': account_receivable.id,
+        })
+        sale_order = self.SudoSaleOrder.create({
+            'partner_id': partner.id,
+            'partner_invoice_id': partner.id,
+            'partner_shipping_id': partner.id,
+        })
+        sale_order_line = self.SudoSaleOrderLine.create({
+            'order_id': sale_order.id,
+            'name': product.name,
+            'product_id': product.id,
+            'product_uom_qty': 0,
+            'product_uom': self.uom_hour.id,
+            'price_unit': product.list_price
+        })
+        sale_order.action_confirm()
+        task = self.SudoProjectTask.search([
+            ('sale_line_id', '=', sale_order_line.id)
+        ])
+        timesheet = self.SudoAccountAnalyticLine.create({
+            'project_id': task.project_id.id,
+            'task_id': task.id,
+            'name': 'Entry #1',
+            'unit_amount': 1,
+            'employee_id': employee.id,
+        })
+        self.assertTrue(timesheet.so_line)
+        self.assertEqual(task.billable_type, 'task_rate')
+
```

Comment:
here it looks to me should end a test

Reply:
Why? I'm always open for suggestions (in code form)